File: caoguodong/Hamming Distance.java
public class Solution {
    public int hammingDistance(int x, int y) {
        int z = ((x|y)&(~(x&y)));
        int result = 0;
        while(z != 0){
            int exr= z % 2;
            z = (z - exr) / 2;
            result+=exr;
        }
        return result;
    }
}

File: songzhanao/Hamming Distance.c
int hammingDistance(int x, int y) {
    //1.找出bit位不同的整数 2.算出这个整数有多少个1
    int z = x^y;
    int d = 0;
    while(z){
        ++d;
        z &= z-1;
    }
    return d;
}


